_validate/createJson.py

```diff
@@ -213,18 +213,30 @@ def main():
 				errorFile.write(f"Validation Errors:\n{manifest.errors}")
 		raise ValueError(f"Invalid manifest file: {manifest.errors}")
 
-	generateJsonFile(
-		manifest=manifest,
-		addonPath=args.file,
-		parentDir=args.parentDir,
-		channel=args.channel,
-		publisher=args.publisher,
-		sourceUrl=args.sourceUrl,
-		url=args.url,
-		licenseName=args.licenseName,
-		# Convert the case --licUrl='' to --licUrl=None
-		licenseUrl=args.licenseUrl if args.licenseUrl else None,
-	)
+	try:
+		generateJsonFile(
+			manifest=manifest,
+			addonPath=args.file,
+			parentDir=args.parentDir,
+			channel=args.channel,
+			publisher=args.publisher,
+			sourceUrl=args.sourceUrl,
+			url=args.url,
+			licenseName=args.licenseName,
+			# Convert the case --licUrl='' to --licUrl=None
+			licenseUrl=args.licenseUrl if args.licenseUrl else None,
+		)
+	except Exception as e:
+		if manifest.errors:
+			if errorFilePath:
+				with open(errorFilePath, "w") as errorFile:
```

Comment:
Do we want to open the log file in append mode?
```suggestion
				with open(errorFilePath, "wa") as errorFile:
```

Reply:
I don't think so, if any other validation errors have occurred the job should have already failed.
Appending means CURLs output, used for debugging, which wouldn't contain validation errors, would be retained in the validation errors file.
CURL outputs to the validation errors file in case the download job fails.